### ISSUE TITLE
Moving required number of digits calculation to fit() - BinaryEncoder and BaseNEncoder

### ISSUE THREAD

**Issue by andrethrill** [scikit-learn-contrib/categorical-encoding#34]
In binary.py, the required number of digits is calculated as such, inside the _binary()_ function:

```
# figure out how many digits we need to represent the classes present
  digits = int(np.ceil(np.log2(len(X[col].unique()))))
```

_binary()_ is called inside the _transform()_ function.

Is there a specific reason why the required number of digits are calculated during _transform()_ instead of _fit()_?

The reason I am asking is because, when performing distributed computation, different partitions of the same array are sent to different machines and the calculated required number of digits would then be different on each machine depending on how many different classes have each partition.

**Comment by wdm0006** [comment]
Yeah, I think this makes sense. One question though, if you have say, 4 categories in the dataset passed to fit(), and determined that these can be encoded with 2 columns, but then pass data with 6 categories to the transform, would getting an exception or imputing the unknown categories with some unknown value be tolerable for your use-case?

**Comment by andrethrill** [comment]
I think that having the option to choose between those two would be appropriate.

**Comment by andrethrill** [comment]
FYI I noticed that *BaseNEncoder* has the same behavior.

If you need any further input from me, feel free to ask

**Comment by andrethrill** [comment]
I've created #35 to fix the problem in binary.py, for my use case (still no option flag). I didn't modify *BaseNEncoder* because I'm not familiar with it

**Comment by wdm0006** [comment]
Fixed by #35 and new issue for basen encoder opened as #36
